subprojects/model-core/src/main/java/org/gradle/internal/metaobject/AbstractDynamicObject.java

```diff
@@ -179,7 +180,20 @@ public MissingMethodException methodMissingException(String name, Object... para
             // Include the display name anyway
             message = String.format("Could not find method %s() for arguments %s on %s.", name, Arrays.toString(params), getDisplayName());
         }
-        return new CustomMessageMissingMethodException(name, publicType, message, params);
+        // https://github.com/apache/groovy/commit/75c068207ba24648ea2d698c520601c6fcf0a45b
+        return new CustomMissingMethodExecutionFailed(name, publicType, message, params);
+    }
+
+    private static class CustomMissingMethodExecutionFailed extends MissingMethodExecutionFailed {
+
+        public CustomMissingMethodExecutionFailed(String name, Class publicType, String message, Object... params) {
+            super(name, publicType, params, false, new CustomMessageMissingMethodException(name, publicType, message));
```

Comment:
The constructor for CustomMessageMissingMethodException is missing `params` here.

I think this is causing the weird method missing exceptions.

Reply:
Yes, that was the case. I recreated the situation in a test here https://github.com/gradle/gradle/pull/12383/commits/222ec9dc6971087f7ccba7f86f3318a50fd0d0e9
I really don't like that test, but it is the simplest reproducer that I could get to so far.